hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -291,15 +291,25 @@ private <BuilderT extends S3BaseClientBuilder<BuilderT, ClientT>, ClientT> void
     if (endpoint != null) {
       checkArgument(!fipsEnabled,
           "%s : %s", ERROR_ENDPOINT_WITH_FIPS, endpoint);
-      builder.endpointOverride(endpoint);
-      // No region was configured, try to determine it from the endpoint.
-      if (region == null) {
-        region = getS3RegionFromEndpoint(parameters.getEndpoint());
-        if (region != null) {
-          origin = "endpoint";
+      if(parameters.getEndpoint().equals(CENTRAL_ENDPOINT)){
```

Comment:
if someone configures `fs.s3a.endpoint` to `s3.amazonaws.com` and sets region in `fs.s3a.endpoint.region` to `eu-west-1`, this code will start ignoring what we have in `fs.s3a.endpoint.region` and just enable cross region for everything. I think this could be risky as people could just have `s.s3a.endpoint` to `s3.amazonaws.com` in their core-site.xml as it doesn't make a difference if you've set your region right.

Reply:
Okay cool, then I will close this PR.